crates/vm/levm/tests/tests.rs

```diff
@@ -3362,3 +3363,146 @@ fn create_on_create() {
     vm.execute();
     assert_eq!(vm.db.accounts.len(), 4);
 }
+
+#[test]
+fn callvalue_op() {
+    let address_that_has_the_code = Address::from_low_u64_be(0x42);
+    let value = U256::from(0x1234);
+
+    let operations = [Operation::Callvalue, Operation::Stop];
+
+    let tx_env = TxEnv {
+        transact_to: TransactTo::Call(address_that_has_the_code),
+        value,
+        ..Default::default()
+    };
+
+    let block_env = BlockEnv::default();
+
+    let mut db = Db::default();
+
+    db.add_account(
+        address_that_has_the_code,
+        Account::default()
+            .with_bytecode(ops_to_bytecde(&operations))
+            .with_balance(U256::MAX),
+    );
+
+    let mut vm = VM::new(tx_env, block_env, db);
+
+    vm.execute();
```

Comment:
Now that `VM::execute` returns an `ExecutionResult`, we could add a check like this:
```suggestion
    assert!(matches!(vm.execute(), ExecutionResult::Success { .. }));
```

Reply:
That is a great idea. I know it is OOC but I realized that in another PR I made some tests that don't look good at all, like:
```
let result = vm.execute();

match result {
  ExecutionResult::Halt {
      reason: VMError::OpcodeNotAllowedInStaticContext,
      gas_used: _,
  } => {
      // Test passes
  }
  _ => panic!("Expected ExecutionResult::Halt with OpcodeNotAllowedInStaticContext"),
}
```
When instead it could've been done in a oneliner:
`assert!(matches!(vm.execute(),ExecutionResult::Halt {reason: VMError::OpcodeNotAllowedInStaticContext, ..}));`

Reply:
Good idea. I added a method `is_success()` to `ExecutionResult`, to use when checking success but not caring about the content.